client/my-sites/stats/stats-module/index.jsx

```diff
@@ -152,66 +154,103 @@ class StatsModule extends Component {
 			'is-refreshing': requesting && ! isLoading,
 		} );
 
+		const shouldHideOldModule = isEnabled( 'stats/new-stats-module-component' ) && path === 'posts';
+
 		return (
-			<div>
-				{ siteId && statType && (
-					<QuerySiteStats statType={ statType } siteId={ siteId } query={ query } />
-				) }
-				{ ! isAllTime && (
-					<SectionHeader
-						className={ headerClass }
-						label={ this.getModuleLabel() }
-						href={ ! summary ? summaryLink : null }
+			<>
+				{ isEnabled( 'stats/new-stats-module-component' ) && ! summary && path === 'posts' && (
+					<StatsListPosts
+						moduleName={ path }
+						data={ data }
+						useShortLabel={ useShortLabel }
+						title={ this.getModuleLabel() }
+						emptyMessage={ moduleStrings.empty }
+						showMore={
+							this.props.showSummaryLink && displaySummaryLink
+								? {
+										url: this.getHref(),
+										label: this.props.translate( 'View all', {
+											context: 'Stats: Button label to expand a panel',
+										} ),
+								  }
+								: undefined
+						}
+						titleURL={ this.getHref() }
+						error={ hasError && <ErrorPanel /> }
+						loader={ isLoading && <StatsModulePlaceholder isLoading={ isLoading } /> }
+						translate={ translate }
 					>
-						{ summary && (
-							<DownloadCsv statType={ statType } query={ query } path={ path } period={ period } />
-						) }
-					</SectionHeader>
+						{ this.props.children }
+					</StatsListPosts>
 				) }
-				<Card compact className={ cardClasses }>
-					{ statType === 'statsFileDownloads' && (
-						<StatsModuleAvailabilityWarning
-							statType={ statType }
-							startOfPeriod={ period && period.startOf }
-						/>
+
+				<div className={ classNames( { 'old-stats-module-hidden': shouldHideOldModule } ) }>
+					{ siteId && statType && (
+						<QuerySiteStats statType={ statType } siteId={ siteId } query={ query } />
 					) }
```

Comment:
Should we avoid rendering the old stats module entirely?

```suggestion
				{ ! shouldHideOldModule && <div>
```